shared_model/backend/protobuf/deserialize_repeated_transactions.hpp

```diff
@@ -0,0 +1,31 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef IROHA_SHARED_MODEL_PROTO_DESERIALIZE_REPEATED_TRANSACTIONS_HPP
+#define IROHA_SHARED_MODEL_PROTO_DESERIALIZE_REPEATED_TRANSACTIONS_HPP
+
+#include "common/result.hpp"
+#include "interfaces/common_objects/transaction_sequence_common.hpp"
+#include "interfaces/iroha_internal/abstract_transport_factory.hpp"
+#include "interfaces/transaction.hpp"
+#include "transaction.pb.h"
+
+namespace shared_model {
+  namespace proto {
+
+    using TransactionFactoryType = interface::AbstractTransportFactory<
+        shared_model::interface::Transaction,
+        iroha::protocol::Transaction>;
+
+    iroha::expected::Result<interface::types::SharedTxsCollectionType,
+                            TransactionFactoryType::Error>
+    deserializeTransactions(
```

Comment:
it seems OK for this PR just to speed up the merge, but apart from that, I think that this approach is wrong, because the task it solves is not constrained to proto transactions. I would solve it with something like
```c++
namespace expected {
  template<typename V, typename E>
  Result<std::vector<V>, E> allValuesOrFirstError(std::vector<Result<V, E>> results) {
    std::vector<V> values;
    for (auto &result : results) {
      if (auto e = resultToOptionalError(result)) {
        return e.value();
      }
      values.emplace_back(resultToOptionalValue(std::move(result)).value());
    }
    return values;
  }
}
```

Reply:
There is no need to generalize until there are at least two places where it can be reused. Templates are cool, but it is better to avoid them if possible

Reply:
when there will be a second place, this piece will most likely not be reused

Reply:
also I think that there are places besides this pr, where the proposed code can be used.